features/admin.push-providers.v1/components/push-providers-grid.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+    GridLayout,
+    ResourceGrid
+} from "@wso2is/react-components";
+import React, { FunctionComponent, MouseEvent, ReactElement, useEffect } from "react";
+import PushProviderCard from "./push-provider-card";
+import "./push-providers-grid.scss";
+
+/**
+ * Props for the Push Providers grid component.
+ */
+export interface PushProvidersGridPropsInterface extends IdentifiableComponentInterface, LoadableComponentInterface {
+    /**
+     * Callback to be fired when a template is selected.
+     */
+    onTemplateSelect: (template: ExtensionTemplateListInterface) => void;
+
+    onTemplatesLoad?: (templates: ExtensionTemplateListInterface[]) => void;
+
+    selectedTemplate: ExtensionTemplateListInterface;
+}
+
+/**
+ * Push Providers grid page.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns Push Providers grid page.
+ */
+const PushProvidersGrid: FunctionComponent<PushProvidersGridPropsInterface> = ({
+    onTemplateSelect,
+    onTemplatesLoad,
+    selectedTemplate,
+    ["data-componentid"]: componentId = "push-providers-grid"
+}: PushProvidersGridPropsInterface): ReactElement => {
+
+    const {
+        templates,
+        categorizedTemplates,
+        isExtensionTemplatesRequestLoading: isPushProviderTemplatesRequestLoading
+    } = useExtensionTemplates();
+
+    const handleTemplateSelection = (
+        e: MouseEvent<HTMLDivElement>,
+        template: ExtensionTemplateListInterface
+    ): void => {
+        if (!template) {
+            return;
+        }
+        // TODO: event publishing is required.
+        onTemplateSelect(template);
+    };
+
+    useEffect(() => {
+        if (templates && onTemplatesLoad) {
+            onTemplatesLoad(templates);
+        }
+    }, [ templates, onTemplatesLoad ]);
+
+
+    return (
+        <GridLayout
+            isLoading={ isPushProviderTemplatesRequestLoading }
+        >
+            {
+                categorizedTemplates.map((category: CategorizedExtensionTemplatesInterface) => {
+                    const templates: ExtensionTemplateListInterface[] = category?.templates;
+
+                    if (templates?.length <= 0) {
+                        return null;
+                    }
+
+                    return (
+                        <div key={ category?.id } className="push-provider-card-group">
+                            <ResourceGrid
+                                isEmpty={
+                                    !Array.isArray(templates)
+                                        || templates.length <= 0
+                                }
+                            >
+                                {
+                                    templates.map(
+                                        (template: ExtensionTemplateListInterface) => {
+                                            return (
+                                                <PushProviderCard
```

Comment:
Can't we reuse the <ResourceGrid.Card> here?

Reply:
@ZiyamSanthosh

Reply:
The `ResouceGrid.Card` component doesn't have support for rendering feature status labels. Hence the usage of a custom component.